src/main.rs

```diff
@@ -1,8 +1,31 @@
-use actix_web::{middleware, web, App, HttpRequest, HttpServer};
+use actix_web::{middleware, web, App, Error, HttpRequest, HttpResponse, HttpServer};
+use serde::{Deserialize, Serialize};
 
-async fn index(req: HttpRequest) -> &'static str {
-    println!("REQ: {req:?}");
-    "Hello world!"
+#[derive(Debug, Serialize, Deserialize)]
+struct MyObj {
+    name: String,
+    number: i32,
+}
+
+/// This handler uses json extractor
+async fn index(item: web::Json<MyObj>) -> HttpResponse {
+    println!("model: {:?}", &item);
+    HttpResponse::Ok().json(item.0) // <- send response
+}
+
+/// This handler uses json extractor with limit
+async fn extract_item(item: web::Json<MyObj>, req: HttpRequest) -> HttpResponse {
+    println!("request: {req:?}");
```

Comment:
Use a logger when running a debug build instead of println?

Reply:
mb, was just the code example that one. Have added our actual code now with the start of item endpoint.